docusaurus/docs/e2e-test-a-plugin/setup-resources.md

```diff
@@ -0,0 +1,83 @@
+
+In the following example, we navigate to a provisioned dashboard. The dashboard has a multi-valued template variable `env` with the values `test` and `prod`. We add a new panel and set a SQL query that refers to the `env` variable. We then spy on the query data request, asserting that it was called with the expanded values associated with the template variable.
+
+```ts
+test('should expand multi-valued variable before calling backend', async ({
+  readProvision,
+  request,
+  page,
+  selectors,
+  grafanaVersion,
+}) => {
+  const dashboard = await readProvisionDashboard({ fileName: 'variable.json' });
+  const dashboardPage = new DashboardPage({ request, page, selectors, grafanaVersion }, dashboard);
+  await dashboardPage.goto();
+  const panelEditPage = await dashboardPage.addPanel();
+  const queryDataSpy = panelEditPage.waitForQueryDataRequest((request) =>
+    (request.postData() ?? '').includes(`select * from dataset where env in ('test', 'prod')"`)
+  );
+  await page.getByLabel('Query').fill('select * from dataset where env in (${env:singlequote})');
+  await panelEditPage.refreshPanel();
+  await expect(await queryDataSpy).toBeTruthy();
+});
+```
+
+## Provision any required resources
+
+You may use [provisioning](https://grafana.com/docs/grafana/latest/administration/provisioning/) to configure resources such as dashboards and data sources.
+
+> If running e2e tests in CI requires provisioning, you may need to remove the `provisioning` folder from your plugin's `.gitignore` file. Be careful not to commit secrets to public repositories. Use [environment variable](https://grafana.com/docs/grafana/latest/administration/provisioning/#using-environment-variables) interpolation for sensitive data.
+
+## Read provisioned files
+
+`@grafana/plugin-e2e` provides fixtures that enables you to read files that you have placed in the `provisioning` folder.
+
+### `readProvisionedDataSource` fixture
+
+The `readProvisionedDataSource` fixture allows you to read a file from your plugin's `provisioning/datasources` folder. This gives you typings and it also allows you to keep sensitive data in one place.
+
+```ts title="configEditor.spec.ts"
+const datasource = readProvisionedDataSource<JsonData, SecureJsonData>({ fileName: 'datasources.yaml' });
+await page.getByLabel('API Key').fill(datasource.secureJsonData.apiKey);
+```
+
+```ts title="queryEditor.spec.ts"
+const datasource = readProvisionedDataSource({ fileName: 'datasources.yaml' });
+await panelEditPage.datasource.set(datasource.name);
+```
+
+### `readProvisionedDashboard` fixture
+
+The `readProvisionedDashboard` fixture allows you to read the content of a dashboard json file. It can be useful when navigating to a provisioned dashboard and you don't want to hard code the dashboard UID.
```

Comment:
any required file path for the dashboard?

Reply:
[Dashboards](https://grafana.com/docs/grafana/latest/administration/provisioning/#dashboards) must be placed in the `provisioning/dashboards` dir, so if the user of this fixture just provides filename it will be enough.

Reply:
for consistency with the DS one above though you could say 
```suggestion
The `readProvisionedDashboard` fixture allows you to read the content of a dashboard JSON file from your`provisioning/dashboards` folder. It can be useful when navigating to a provisioned dashboard and you don't want to hard code the dashboard UID.
```

I think we should either capitalize JSON or say `dashboard.json`